test/run/run-e2e-tests.js

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env node
+
+const { promisify } = require( 'util' );
+const { initApp } = require( './init_app' );
+const { initLogs } = require( './init_logs' );
+const { initTests } = require( './init_tests' );
+const { initDriver } = require( './init_driver' );
+const { checkCredentials } = require( './checks' );
+
+let app;
+let driver;
+
+const delay = promisify( setTimeout );
+
+// Handle both user-initiated (SIGINT) and normal termination.
+process.on( 'SIGINT', function() {
+    handleExit();
+    process.exit();
+} );
+
+process.on( 'exit', handleExit );
+function handleExit() {
+    if ( driver ) {
+        driver.kill();
+    }
+    if ( app ) {
+        app.kill();
+    }
+}
+
+async function run() {
+    try {
+        checkCredentials();
+
+        const timestamp = ( new Date() ).toJSON().replace( /:/g, '-' );
+        const { appLog, driverLog } = initLogs( timestamp );
+
+        app = initApp( appLog );
+        await delay( 5000 );
+
+        driver = initDriver( driverLog );
+
+        await initTests();
+    }
+    catch ( err ) {
+        console.error( err );
+    }
+    finally {
+        process.exit();
+    }
+}
```

Comment:
just because the entire function body is a `try` maybe it's worth splitting? probably fine as-is, but feels a little silly having a big function body inside a `try`

```js
async function run_unsafe() {
  …
}

async function run() {
  try {
    run_unsafe();
  } catch (err) {
    console.error( err );
  } finally {
    process.exit()
  }
}
```

Reply:
In this specific case, the entirety of the logic directly encapsulated within the `try` block is evident at a glance, and further encapsulation in a wrapper method is unnecessary (and IMO adds an unnecessary level of indirection, and could be even be confusing). My preference here is to leave it as-is.